### DISCUSSION TITLE
Erkennung des HB-RF-USB-TK im Docker unter Synology NAS/DSM

### DISCUSSION THREAD

**Discussion by Mikei386** [OpenCCU/OpenCCU#1126] [🇩🇪 Fragen&Antworten] [answered]
Hallo,

ersteinmal Spitzenklasse das es jetzt was für Docker gibt.

Ich habe es bisher auf einer Synology in einer VM mit der HB-RF-USB-TK am laufen.
Da es auf der Synology schwer ist, direkt im System etwas zu installieren wäre meine Frage ob es möglich ist,
die Erkennung der HB-RF-USB-TK im Container zu ermöglichen, indem der USB-Bus im Container gemundet wird:

```-v /dev/bus/usb:/dev/bus/usb```

Ich habe weitere Container, die mittels diesem Befehl auf USB-Geräte zugreifen können, z.B.: SDR-USB-Stick.
Ein weiterer Container nutzt ein USB Gerät mittels: `--device=/dev/bus/usb`

Gibt es da bereits eine Möglichkeit?

Danke 
Michael

**Comment by jens-maus** [comment, marked as answer]
> Ich habe es bisher auf einer Synology in einer VM mit der HB-RF-USB-TK am laufen.
> Da es auf der Synology schwer ist, direkt im System etwas zu installieren wäre meine Frage ob es möglich ist,
> die Erkennung der HB-RF-USB-TK im Container zu ermöglichen, indem der USB-Bus im Container gemundet wird:
> [...]
> Gibt es da bereits eine Möglichkeit?

Wenn es nur das USB Gerät des HB-RF-USB wäre, welches weitergeleitet werden müsste, wäre das eine Option, ja. Allerdings ist es ja so das das eigentliche Funkmodul das auf dem HB-RF-USB sitzt selbst ja noch Linux Kerneltreiber braucht um korrekt zu funktionieren bzw. angesteuert zu werden. Und da eine Docker-Umgebung ja eine Paravirtualisierung ist wird in dem Falle der Kerneltreiber im Hostsystem benötigt. D.h. man müsste erst einmal innerhalb der Linux-Umgebung des DSM Betriebssystemes von Synology den Kerneltreiber verfügbar machen.

Das steht momentan nicht nur nicht auf der Agenda, sondern wird auch wegen der vielen verschiedenen und teils veralteten Kernelversionen die Synology einsetzt ein mitunter unmögliches Unterfangen. Ergo ist es weiterhin die beste und einfachste Option bei Einsatz einer Synology Diskstation oder QNAP, etc. besser weiterhin auf eine volle Virtualisierung via OVA Variante zu setzen.

**Reply by Mikei386** [reply]
Vielen Dank für die Antwort.

Dann lasse ich es weiter in der VM laufen. Ist ja auch eine praktische Sache.

Vielen Dank für die Mühe und viele Grüße
